### PR TITLE
tools: Add script to re-run queued workflows

### PR DESCRIPTION
**Pull request by seemethere**
Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

File: tools/self-hosted-runner-utils/rerun_queued_workflows.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import os
+import argparse
+from datetime import datetime, timedelta
+
+from github import Github
+
+
+def pretty_time_delta(seconds):
```
Comment (malfet) [inline tools/self-hosted-runner-utils/rerun_queued_workflows.py:8]:
```suggestion
def pretty_time_delta(seconds: float) -> str:
```

**Comment by seemethere** [comment]
<details>
<summary> Current dry run output </summary>

```
.venv ❯ python rerun_queued_workflows.py --dry-run pytorch/pytorch
+ (Queued for   6h59m46s) https://github.com/pytorch/pytorch/actions/runs/1291967011...skipped (DRY_RUN)
+ (Queued for   6h59m46s) https://github.com/pytorch/pytorch/actions/runs/1291967017...skipped (DRY_RUN)
+ (Queued for   6h59m46s) https://github.com/pytorch/pytorch/actions/runs/1291967008...skipped (DRY_RUN)
+ (Queued for   6h59m45s) https://github.com/pytorch/pytorch/actions/runs/1291967005...skipped (DRY_RUN)
+ (Queued for   6h59m45s) https://github.com/pytorch/pytorch/actions/runs/1291966998...skipped (DRY_RUN)
+ (Queued for   6h59m45s) https://github.com/pytorch/pytorch/actions/runs/1291966997...skipped (DRY_RUN)
+ (Queued for   6h59m45s) https://github.com/pytorch/pytorch/actions/runs/1291967001...skipped (DRY_RUN)
+ (Queued for   6h59m45s) https://github.com/pytorch/pytorch/actions/runs/1291967003...skipped (DRY_RUN)
+ (Queued for    6h58m3s) https://github.com/pytorch/pytorch/actions/runs/1291960866...skipped (DRY_RUN)
+ (Queued for    6h58m4s) https://github.com/pytorch/pytorch/actions/runs/1291960867...skipped (DRY_RUN)
+ (Queued for    6h58m4s) https://github.com/pytorch/pytorch/actions/runs/1291960865...skipped (DRY_RUN)
+ (Queued for   6h57m41s) https://github.com/pytorch/pytorch/actions/runs/1291960875...skipped (DRY_RUN)
+ (Queued for   6h58m13s) https://github.com/pytorch/pytorch/actions/runs/1291960870...skipped (DRY_RUN)
+ (Queued for    6h58m3s) https://github.com/pytorch/pytorch/actions/runs/1291960860...skipped (DRY_RUN)
+ (Queued for   6h58m11s) https://github.com/pytorch/pytorch/actions/runs/1291960864...skipped (DRY_RUN)
+ (Queued for   6h57m48s) https://github.com/pytorch/pytorch/actions/runs/1291960430...skipped (DRY_RUN)
+ (Queued for   6h54m57s) https://github.com/pytorch/pytorch/actions/runs/1291951052...skipped (DRY_RUN)
+ (Queued for    6h55m0s) https://github.com/pytorch/pytorch/actions/runs/1291951044...skipped (DRY_RUN)
+ (Queued for    6h55m0s) https://github.com/pytorch/pytorch/actions/runs/1291951049...skipped (DRY_RUN)
+ (Queued for    6h55m7s) https://github.com/pytorch/pytorch/actions/runs/1291951047...skipped (DRY_RUN)
+ (Queued for    6h55m0s) https://github.com/pytorch/pytorch/actions/runs/1291951046...skipped (DRY_RUN)
+ (Queued for    6h55m4s) https://github.com/pytorch/pytorch/actions/runs/1291951054...skipped (DRY_RUN)
+ (Queued for    6h55m4s) https://github.com/pytorch/pytorch/actions/runs/1291951043...skipped (DRY_RUN)
+ (Queued for   6h54m28s) https://github.com/pytorch/pytorch/actions/runs/1291951048...skipped (DRY_RUN)
+ (Queued for   6h54m55s) https://github.com/pytorch/pytorch/actions/runs/1291951017...skipped (DRY_RUN)
+ (Queued for   6h54m56s) https://github.com/pytorch/pytorch/actions/runs/1291951006...skipped (DRY_RUN)
+ (Queued for   6h54m55s) https://github.com/pytorch/pytorch/actions/runs/1291951015...skipped (DRY_RUN)
+ (Queued for   6h54m58s) https://github.com/pytorch/pytorch/actions/runs/1291951005...skipped (DRY_RUN)
+ (Queued for   6h54m27s) https://github.com/pytorch/pytorch/actions/runs/1291951008...skipped (DRY_RUN)
+ (Queued for    6h55m2s) https://github.com/pytorch/pytorch/actions/runs/1291951013...skipped (DRY_RUN)
+ (Queued for   6h54m25s) https://github.com/pytorch/pytorch/actions/runs/1291951008...skipped (DRY_RUN)
+ (Queued for   6h54m54s) https://github.com/pytorch/pytorch/actions/runs/1291951016...skipped (DRY_RUN)
+ (Queued for   6h51m23s) https://github.com/pytorch/pytorch/actions/runs/1291938752...skipped (DRY_RUN)
+ (Queued for   6h51m22s) https://github.com/pytorch/pytorch/actions/runs/1291938738...skipped (DRY_RUN)
+ (Queued for   6h51m12s) https://github.com/pytorch/pytorch/actions/runs/1291938748...skipped (DRY_RUN)
+ (Queued for   6h51m21s) https://github.com/pytorch/pytorch/actions/runs/1291938737...skipped (DRY_RUN)
+ (Queued for   6h51m11s) https://github.com/pytorch/pytorch/actions/runs/1291938746...skipped (DRY_RUN)
+ (Queued for   6h51m26s) https://github.com/pytorch/pytorch/actions/runs/1291938742...skipped (DRY_RUN)
+ (Queued for   6h51m20s) https://github.com/pytorch/pytorch/actions/runs/1291938745...skipped (DRY_RUN)
+ (Queued for   6h51m25s) https://github.com/pytorch/pytorch/actions/runs/1291938741...skipped (DRY_RUN)
+ (Queued for   6h50m47s) https://github.com/pytorch/pytorch/actions/runs/1291938751...skipped (DRY_RUN)
+ (Queued for   6h51m23s) https://github.com/pytorch/pytorch/actions/runs/1291938749...skipped (DRY_RUN)
+ (Queued for   6h51m21s) https://github.com/pytorch/pytorch/actions/runs/1291938750...skipped (DRY_RUN)
+ (Queued for   6h51m20s) https://github.com/pytorch/pytorch/actions/runs/1291938744...skipped (DRY_RUN)
+ (Queued for    6h51m7s) https://github.com/pytorch/pytorch/actions/runs/1291938743...skipped (DRY_RUN)
+ (Queued for    6h51m9s) https://github.com/pytorch/pytorch/actions/runs/1291938739...skipped (DRY_RUN)
+ (Queued for   6h42m21s) https://github.com/pytorch/pytorch/actions/runs/1291915766...skipped (DRY_RUN)
+ (Queued for   6h42m24s) https://github.com/pytorch/pytorch/actions/runs/1291915762...skipped (DRY_RUN)
+ (Queued for   6h41m56s) https://github.com/pytorch/pytorch/actions/runs/1291915758...skipped (DRY_RUN)
+ (Queued for   6h42m24s) https://github.com/pytorch/pytorch/actions/runs/1291915769...skipped (DRY_RUN)
+ (Queued for   6h42m21s) https://github.com/pytorch/pytorch/actions/runs/1291915767...skipped (DRY_RUN)
+ (Queued for   6h42m21s) https://github.com/pytorch/pytorch/actions/runs/1291915760...skipped (DRY_RUN)
+ (Queued for   6h42m16s) https://github.com/pytorch/pytorch/actions/runs/1291915759...skipped (DRY_RUN)
+ (Queued for   6h41m20s) https://github.com/pytorch/pytorch/actions/runs/1291913028...skipped (DRY_RUN)
+ (Queued for   6h39m25s) https://github.com/pytorch/pytorch/actions/runs/1291907213...skipped (DRY_RUN)
+ (Queued for   6h39m20s) https://github.com/pytorch/pytorch/actions/runs/1291907207...skipped (DRY_RUN)
+ (Queued for   6h39m20s) https://github.com/pytorch/pytorch/actions/runs/1291907205...skipped (DRY_RUN)
+ (Queued for   6h38m51s) https://github.com/pytorch/pytorch/actions/runs/1291907202...skipped (DRY_RUN)
+ (Queued for   6h39m18s) https://github.com/pytorch/pytorch/actions/runs/1291907212...skipped (DRY_RUN)
+ (Queued for   6h39m16s) https://github.com/pytorch/pytorch/actions/runs/1291907204...skipped (DRY_RUN)
+ (Queued for   6h39m23s) https://github.com/pytorch/pytorch/actions/runs/1291907213...skipped (DRY_RUN)
+ (Queued for   6h39m16s) https://github.com/pytorch/pytorch/actions/runs/1291907212...skipped (DRY_RUN)
+ (Queued for   6h39m14s) https://github.com/pytorch/pytorch/actions/runs/1291907204...skipped (DRY_RUN)
+ (Queued for   6h39m18s) https://github.com/pytorch/pytorch/actions/runs/1291907207...skipped (DRY_RUN)
+ (Queued for   6h39m18s) https://github.com/pytorch/pytorch/actions/runs/1291907205...skipped (DRY_RUN)
+ (Queued for   6h36m41s) https://github.com/pytorch/pytorch/actions/runs/1291900136...skipped (DRY_RUN)
+ (Queued for   6h36m43s) https://github.com/pytorch/pytorch/actions/runs/1291900144...skipped (DRY_RUN)
+ (Queued for   6h36m41s) https://github.com/pytorch/pytorch/actions/runs/1291900143...skipped (DRY_RUN)
+ (Queued for   6h36m44s) https://github.com/pytorch/pytorch/actions/runs/1291900135...skipped (DRY_RUN)
+ (Queued for    6h36m9s) https://github.com/pytorch/pytorch/actions/runs/1291900142...skipped (DRY_RUN)
+ (Queued for   6h36m41s) https://github.com/pytorch/pytorch/actions/runs/1291900141...skipped (DRY_RUN)
+ (Queued for   6h33m56s) https://github.com/pytorch/pytorch/actions/runs/1291892217...skipped (DRY_RUN)
+ (Queued for   6h33m54s) https://github.com/pytorch/pytorch/actions/runs/1291892222...skipped (DRY_RUN)
+ (Queued for   6h33m52s) https://github.com/pytorch/pytorch/actions/runs/1291892225...skipped (DRY_RUN)
+ (Queued for   6h33m48s) https://github.com/pytorch/pytorch/actions/runs/1291892219...skipped (DRY_RUN)
+ (Queued for   6h33m54s) https://github.com/pytorch/pytorch/actions/runs/1291892213...skipped (DRY_RUN)
+ (Queued for   6h33m55s) https://github.com/pytorch/pytorch/actions/runs/1291892234...skipped (DRY_RUN)
+ (Queued for   6h33m55s) https://github.com/pytorch/pytorch/actions/runs/1291892224...skipped (DRY_RUN)
+ (Queued for   6h33m50s) https://github.com/pytorch/pytorch/actions/runs/1291892223...skipped (DRY_RUN)
+ (Queued for   6h33m57s) https://github.com/pytorch/pytorch/actions/runs/1291892220...skipped (DRY_RUN)
+ (Queued for   6h34m51s) https://github.com/pytorch/pytorch/actions/runs/1291892218...skipped (DRY_RUN)
+ (Queued for   6h33m25s) https://github.com/pytorch/pytorch/actions/runs/1291892227...skipped (DRY_RUN)
+ (Queued for   6h32m57s) https://github.com/pytorch/pytorch/actions/runs/1291889724...skipped (DRY_RUN)
+ (Queued for   6h32m56s) https://github.com/pytorch/pytorch/actions/runs/1291889726...skipped (DRY_RUN)
+ (Queued for    6h54m3s) https://github.com/pytorch/pytorch/actions/runs/1291876930...skipped (DRY_RUN)
+ (Queued for   6h50m46s) https://github.com/pytorch/pytorch/actions/runs/1291876937...skipped (DRY_RUN)
+ (Queued for   6h56m43s) https://github.com/pytorch/pytorch/actions/runs/1291876941...skipped (DRY_RUN)
+ (Queued for   6h52m31s) https://github.com/pytorch/pytorch/actions/runs/1291876942...skipped (DRY_RUN)
+ (Queued for   6h42m53s) https://github.com/pytorch/pytorch/actions/runs/1291801177...skipped (DRY_RUN)
+ (Queued for   6h49m33s) https://github.com/pytorch/pytorch/actions/runs/1291801171...skipped (DRY_RUN)
+ (Queued for   6h36m20s) https://github.com/pytorch/pytorch/actions/runs/1291797416...skipped (DRY_RUN)
+ (Queued for   6h35m16s) https://github.com/pytorch/pytorch/actions/runs/1291792437...skipped (DRY_RUN)
+ (Queued for   6h34m49s) https://github.com/pytorch/pytorch/actions/runs/1291792331...skipped (DRY_RUN)
+ (Queued for    6h57m1s) https://github.com/pytorch/pytorch/actions/runs/1291780365...skipped (DRY_RUN)
+ (Queued for   6h31m16s) https://github.com/pytorch/pytorch/actions/runs/1291780363...skipped (DRY_RUN)
+ (Queued for   6h27m54s) https://github.com/pytorch/pytorch/actions/runs/1291739368...skipped (DRY_RUN)
+ (Queued for   6h27m37s) https://github.com/pytorch/pytorch/actions/runs/1291739364...skipped (DRY_RUN)
+ (Queued for   6h26m14s) https://github.com/pytorch/pytorch/actions/runs/1291739365...skipped (DRY_RUN)
+ (Queued for   6h46m57s) https://github.com/pytorch/pytorch/actions/runs/1291739349...skipped (DRY_RUN)
+ (Queued for   6h19m11s) https://github.com/pytorch/pytorch/actions/runs/1291724444...skipped (DRY_RUN)
+ (Queued for   6h50m21s) https://github.com/pytorch/pytorch/actions/runs/1291724442...skipped (DRY_RUN)
+ (Queued for   6h15m15s) https://github.com/pytorch/pytorch/actions/runs/1291703428...skipped (DRY_RUN)
+ (Queued for   6h42m36s) https://github.com/pytorch/pytorch/actions/runs/1291703347...skipped (DRY_RUN)
+ (Queued for    6h5m15s) https://github.com/pytorch/pytorch/actions/runs/1291703346...skipped (DRY_RUN)
+ (Queued for   6h43m34s) https://github.com/pytorch/pytorch/actions/runs/1291703352...skipped (DRY_RUN)
+ (Queued for   6h14m47s) https://github.com/pytorch/pytorch/actions/runs/1291703354...skipped (DRY_RUN)
+ (Queued for    6h4m55s) https://github.com/pytorch/pytorch/actions/runs/1291686662...skipped (DRY_RUN)
+ (Queued for    6h5m56s) https://github.com/pytorch/pytorch/actions/runs/1291686672...skipped (DRY_RUN)
+ (Queued for    6h37m9s) https://github.com/pytorch/pytorch/actions/runs/1291686661...skipped (DRY_RUN)
+ (Queued for    6h4m46s) https://github.com/pytorch/pytorch/actions/runs/1291686664...skipped (DRY_RUN)
+ (Queued for   6h33m42s) https://github.com/pytorch/pytorch/actions/runs/1291667103...skipped (DRY_RUN)
+ (Queued for   6h47m50s) https://github.com/pytorch/pytorch/actions/runs/1291666804...skipped (DRY_RUN)
+ (Queued for   6h33m42s) https://github.com/pytorch/pytorch/actions/runs/1291666792...skipped (DRY_RUN)
+ (Queued for    6h15m3s) https://github.com/pytorch/pytorch/actions/runs/1291586064...skipped (DRY_RUN)
+ (Queued for    6h54m6s) https://github.com/pytorch/pytorch/actions/runs/1291586113...skipped (DRY_RUN)
+ (Queued for   6h31m50s) https://github.com/pytorch/pytorch/actions/runs/1291586069...skipped (DRY_RUN)
+ (Queued for   6h57m11s) https://github.com/pytorch/pytorch/actions/runs/1291586109...skipped (DRY_RUN)
+ (Queued for   5h50m22s) https://github.com/pytorch/pytorch/actions/runs/1291577076...skipped (DRY_RUN)
+ (Queued for   6h30m42s) https://github.com/pytorch/pytorch/actions/runs/1291577073...skipped (DRY_RUN)
+ (Queued for   5h50m56s) https://github.com/pytorch/pytorch/actions/runs/1291515071...skipped (DRY_RUN)
+ (Queued for   5h40m59s) https://github.com/pytorch/pytorch/actions/runs/1291472285...skipped (DRY_RUN)
+ (Queued for    6h4m58s) https://github.com/pytorch/pytorch/actions/runs/1291472296...skipped (DRY_RUN)
+ (Queued for   6h41m41s) https://github.com/pytorch/pytorch/actions/runs/1291458163...skipped (DRY_RUN)
+ (Queued for   6h45m24s) https://github.com/pytorch/pytorch/actions/runs/1291458171...skipped (DRY_RUN)
+ (Queued for   6h25m26s) https://github.com/pytorch/pytorch/actions/runs/1291426840...skipped (DRY_RUN)
+ (Queued for   4h42m27s) https://github.com/pytorch/pytorch/actions/runs/1291378144...skipped (DRY_RUN)
+ (Queued for   5h24m42s) https://github.com/pytorch/pytorch/actions/runs/1291378150...skipped (DRY_RUN)
+ (Queued for   4h43m15s) https://github.com/pytorch/pytorch/actions/runs/1291366524...skipped (DRY_RUN)
+ (Queued for   5h26m26s) https://github.com/pytorch/pytorch/actions/runs/1291366528...skipped (DRY_RUN)
+ (Queued for   6h21m50s) https://github.com/pytorch/pytorch/actions/runs/1291299705...skipped (DRY_RUN)
+ (Queued for   6h25m19s) https://github.com/pytorch/pytorch/actions/runs/1291299701...skipped (DRY_RUN)
+ (Queued for   4h25m46s) https://github.com/pytorch/pytorch/actions/runs/1291136434...skipped (DRY_RUN)
+ (Queued for   4h54m47s) https://github.com/pytorch/pytorch/actions/runs/1291136439...skipped (DRY_RUN)
+ (Queued for   4h24m22s) https://github.com/pytorch/pytorch/actions/runs/1291128326...skipped (DRY_RUN)
+ (Queued for   4h58m58s) https://github.com/pytorch/pytorch/actions/runs/1291128336...skipped (DRY_RUN)
+ (Queued for   4h23m20s) https://github.com/pytorch/pytorch/actions/runs/1291100441...skipped (DRY_RUN)
+ (Queued for   4h44m58s) https://github.com/pytorch/pytorch/actions/runs/1291100444...skipped (DRY_RUN)
+ (Queued for   4h20m57s) https://github.com/pytorch/pytorch/actions/runs/1290996441...skipped (DRY_RUN)
+ (Queued for   4h44m18s) https://github.com/pytorch/pytorch/actions/runs/1290996445...skipped (DRY_RUN)
+ (Queued for   4h11m41s) https://github.com/pytorch/pytorch/actions/runs/1290934822...skipped (DRY_RUN)
+ (Queued for   4h40m48s) https://github.com/pytorch/pytorch/actions/runs/1290934841...skipped (DRY_RUN)
+ (Queued for   4h10m17s) https://github.com/pytorch/pytorch/actions/runs/1290929285...skipped (DRY_RUN)
+ (Queued for   4h35m12s) https://github.com/pytorch/pytorch/actions/runs/1290929284...skipped (DRY_RUN)
+ (Queued for   4h10m11s) https://github.com/pytorch/pytorch/actions/runs/1290927070...skipped (DRY_RUN)
+ (Queued for   3h54m44s) https://github.com/pytorch/pytorch/actions/runs/1290927083...skipped (DRY_RUN)
+ (Queued for   4h32m29s) https://github.com/pytorch/pytorch/actions/runs/1290799807...skipped (DRY_RUN)
+ (Queued for   4h56m31s) https://github.com/pytorch/pytorch/actions/runs/1290799799...skipped (DRY_RUN)
+ (Queued for   4h25m31s) https://github.com/pytorch/pytorch/actions/runs/1290605535...skipped (DRY_RUN)
+ (Queued for   3h46m40s) https://github.com/pytorch/pytorch/actions/runs/1290605533...skipped (DRY_RUN)
+ (Queued for     24m24s) https://github.com/pytorch/pytorch/actions/runs/1290173807...skipped (DRY_RUN)
+ (Queued for      10m8s) https://github.com/pytorch/pytorch/actions/runs/1290089318...skipped (DRY_RUN)
+ (Queued for   6h30m43s) https://github.com/pytorch/pytorch/actions/runs/1289923901...skipped (DRY_RUN)
+ (Queued for   4h17m35s) https://github.com/pytorch/pytorch/actions/runs/1289784445...skipped (DRY_RUN)
+ (Queued for   5h24m12s) https://github.com/pytorch/pytorch/actions/runs/1289323817...skipped (DRY_RUN)

```

</details>

File: tools/self-hosted-runner-utils/rerun_queued_workflows.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    # taken from https://gist.github.com/thatalextaylor/7408395
+    sign_string = '-' if seconds < 0 else ''
```
Comment (malfet) [inline tools/self-hosted-runner-utils/rerun_queued_workflows.py:10]: Aren't we supposed to use double quotes, per https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings
```suggestion
    sign_string = "-" if seconds < 0 else ""
```

File: tools/self-hosted-runner-utils/rerun_queued_workflows.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    hours, seconds = divmod(seconds, 3600)
+    minutes, seconds = divmod(seconds, 60)
+    if days > 0:
+        return '%s%dd%dh%dm%ds' % (sign_string, days, hours, minutes, seconds)
+    elif hours > 0:
+        return '%s%dh%dm%ds' % (sign_string, hours, minutes, seconds)
+    elif minutes > 0:
+        return '%s%dm%ds' % (sign_string, minutes, seconds)
+    else:
+        return '%s%ds' % (sign_string, seconds)
```
Comment (malfet) [inline tools/self-hosted-runner-utils/rerun_queued_workflows.py:22]: `f-string` please
```suggestion
    if days > 0:
        return f"{sign_string}{int(days)}d{int(hours)}h{int(minutes)}m{int(seconds)}s"
    elif hours > 0:
        return f"{sign_string}{int(hours)}h{int(minutes)}m{int(seconds)}s"
    elif minutes > 0:
        return f"{sign_string}{int(minutes)}m{int(seconds)}s"
    else:
        return f"{sign_string}{int(seconds)}s"
```

**Review by malfet — APPROVED** [review]

**Comment by seemethere** [comment]
This might actually not really be needed so going to close this PR

**Pull request closed without merging**